src/zkevm_specs/evm/typing.py

```diff
@@ -1,7 +1,53 @@
 from typing import Iterator, Optional, Sequence, Union
 
-from ..util import U64, U160, U256, Array4, RLCStore, keccak256
-from .table import TxContextFieldTag
+from ..util import U64, U160, U256, Array3, Array4, RLCStore, keccak256
+from .table import BlockContextFieldTag, TxContextFieldTag
+
+
+class Block:
+    coinbase: U160
+    gas_limit: U64
+    block_number: U256
+    time: U256
+    difficulty: U256
+    base_fee: U256
+
+    # previous_block_hashes contains at most previous 256 block hashes, where
+    # the lastest one is at previous_block_hashes[-1].
+    previous_block_hashes: Sequence[U256]
+
+    def __init__(
+        self,
+        coinbase: U160 = 0x10,
+        gas_limit: U64 = int(15e6),
+        block_number: U256 = 0,
+        time: U256 = 0,
+        difficulty: U256 = 0,
+        base_fee: U256 = int(1e9),
+        previous_block_hashes: Sequence[U256] = [],
+    ) -> None:
+        assert len(previous_block_hashes) <= 256
+
+        self.coinbase = coinbase
+        self.gas_limit = gas_limit
+        self.block_number = block_number
+        self.time = time
+        self.difficulty = difficulty
+        self.base_fee = base_fee
+        self.previous_block_hashes = previous_block_hashes
+
+    def table_assignments(self, rlc_store: RLCStore) -> Sequence[Array3]:
+        return [
+            (BlockContextFieldTag.Coinbase, 0, self.coinbase),
```

Comment:
use self.block_number here ?  `(BlockContextFieldTag.Coinbase, 0, self.block_number), 
(BlockContextFieldTag.GasLimit, 0, self.block_number),` the following does the same .

Reply:
The `block_number_or_zero` is only meaningful for `BlockContextFieldTag.BlockHash`, and will be `0` for other tags. The reason to have `block_number_or_zero` is because prover can index one of 256 block hashes by stack value, so we build such friendly lookup table for it to use.

Reply:
I guess you assume block table  only have one block, my first idea is table look like:    
| block_number |tag   | value|
| -------    |   ---- | -----  |
| 1000704| hash    | 0x29..38 |
| 1000704| coinbase  | 0x5B3..C4|. 

which can hold multiple blocks,   it would be useful for aggregate block proofs, let me know your thinking here

Reply:
Yes, I assume in the same proof we are in the same block. If in the future we need to verify multiple block in a proof, I think we only need to add another `block_number` column just like what you propose.

Reply:
OK, make sense to me for now .